lib/mu_search/index_definition.rb

```diff
@@ -82,10 +82,10 @@ def self.from_json_config(all_definitions)
 
     def self.ensure_uuid_in_properties properties
       properties["uuid"] = ["http://mu.semte.ch/vocabularies/core/uuid"] unless properties.key?("uuid")
-      properties.collect do |property_spec|
-        if property_spec.is_a?(Hash) and property_spec.key?("properties")
+      properties.each do |(_key, value)|
+        if value.is_a?(Hash) and value.key?("properties")
```

Comment:
I think you can just check the property definition here  (admittedly the type should move to a constant)
```
        property_definition = PropertyDefinition.from_json_config(key, value)
        if property_defintion.type ==  "nested"
```

Reply:
Reworked the code to make better use of the property definitions